test/unit/expiration-date.js

```diff
@@ -144,7 +144,8 @@ describe('expirationDate validates', function () {
         ['01201', {isValid: false, isPotentiallyValid: true, month: null, year: null}],
         ['01211', {isValid: false, isPotentiallyValid: false, month: null, year: null}],
         ['01199', {isValid: false, isPotentiallyValid: false, month: null, year: null}],
-        ['01199', {isValid: false, isPotentiallyValid: false, month: null, year: null}]
+        ['01199', {isValid: false, isPotentiallyValid: false, month: null, year: null}],
+        ['122', {isValid: false, isPotentiallyValid: true, month: null, year: null}]
```

Comment:
This fundamentally changes how expiration dates are parsed. Without your changes, this is read as `01 / 22`. With your changes, it's read as invalid.

If you can give a good case for it, we may still make this change, but it would necessitate a major version bump.

Reply:
> This fundamentally changes how expiration dates are parsed

No, I just brought in an extra layer of testing for a value that is less than three, because you agree that the value of 122 is not 01 month and 2? a year? after all, we write from left to right

> Without your changes, this is read as 01 / 22

I keep only the raw data in the model so it's easy to operate.

Under mask '{{99}} / {{99}}' we see that the value should be four-digit and if the user entered 122 then these data were parsed incorrectly i.e. 122 was regarded as 1 month and 22 years and were considered valid which is not true

Reply:
> No, I just brought in an extra layer of testing for a value that is less than three, because you agree that the value of 122 is not 01 month and 2? a year? after all, we write from left to right

On master, `122` is `01/22`.

```js
var cv = require('card-validator');

cv.expirationDate('122'); // {month: '1', year: '22', isValid: true, isPotentiallyValid: true}
```

>Under mask '{{99}} / {{99}}' we see that the value should be four-digit and if the user entered 122 then these data were parsed incorrectly i.e. 122 was regarded as 1 month and 22 years and were considered valid which is not true

The card validator library has no knowledge of how input formatting in restricted input works, it just processes whatever raw data is given to it. You can see here in `braintree-web` that the decision for how to format the input before sending it to card validator is made there, it shouldn't be a concern of the `card-validator` library.

https://github.com/braintree/braintree-web/blob/31f490141a3e5cc17634d3e62dfde06451dfc748/src/hosted-fields/internal/components/expiration-date-input.js#L21-L27

Reply:
It begs the parameter with which you can control the result of the parser, what do you think about it ?